test/csuite/wt8057_compact_stress/main.c

```diff
@@ -0,0 +1,381 @@
+
+    /* Kill the child process. */
+    printf("Kill child\n");
+    testutil_checksys(kill(pid, SIGKILL) != 0);
+    testutil_checksys(waitpid(pid, &status, 0) == -1);
+
+    /* Reopen the connection and verify that the tables match each other. */
+    testutil_check(wiredtiger_open(home, &event_handler, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    verify_tables(session);
+
+    /* Clean-up. */
+    testutil_check(session->close(session, NULL));
+    testutil_check(conn->close(conn, NULL));
+    testutil_cleanup(opts);
+
+    return (EXIT_SUCCESS);
+}
+
+static void
+run_test(const char *home)
+{
+
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    FILE *fp;
+    uint64_t file_sz_after, file_sz_before;
+    bool first_ckpt;
+    int i;
+    int key_range_start;
+    time_t t;
+
+    first_ckpt = false;
+    srand((u_int)time(&t));
+
+    testutil_check(wiredtiger_open(home, &event_handler, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, uri1, table_config));
+    testutil_check(session->create(session, uri2, table_config));
+
+    populate(session, 0, NUM_RECORDS);
+
+    /*
+     * Although we are repeating the steps 40 times, we expect parent process will kill us way
+     * before than that.
+     */
+    for (i = 0; i < 40; i++) {
+
+        printf("Running Loop: %i\n", i + 1);
+        testutil_check(session->checkpoint(session, NULL));
+        /*
+         * Create the checkpoint file so that the parent process knows at least one checkpoint has
+         * finished and can start its timer.
+         */
+        if (!first_ckpt) {
+            testutil_checksys((fp = fopen(ckpt_file, "w")) == NULL);
+            testutil_checksys(fclose(fp) != 0);
+            first_ckpt = true;
+        }
+
+        /*
+         * Remove 1/3 of data from the middle of the key range to let compact relocate blocks from
+         * the end of the file.
+         */
+        key_range_start = rand() % (((NUM_RECORDS * 2) / 3) - 1);
+        remove_records(session, uri1, key_range_start, key_range_start + NUM_RECORDS / 3);
+        remove_records(session, uri2, key_range_start, key_range_start + NUM_RECORDS / 3);
+
+        file_sz_before = get_file_size(session, uri1);
+
+        testutil_check(session->compact(session, uri1, NULL));
+
+        file_sz_after = get_file_size(session, uri1);
+
+        /* Check if there's at least 10% compaction. */
+        printf(" - Compressed file size MB: %f\n - Original file size MB: %f\n",
+          file_sz_after / (1024.0 * 1024), file_sz_before / (1024.0 * 1024));
+
+        /* Put the deleted records back. */
+        populate(session, key_range_start, key_range_start + NUM_RECORDS / 3);
+
+        sleep(1);
+    }
+
+    /* Clean-up. We do not expect test to reach here. */
```

Comment:
Should we assert here instead of closing the session and connection?

Reply:
Added an assert in 21c4606.